src/tools/utils.ts

```diff
@@ -45,6 +47,20 @@ export function keywordReplace(input: string, mappings: KeywordMappings): string
   return input;
 }
 
+// wrapArrayReplaceMarkersInStrings will wrap array replacement markers in quotes.
+// This is necessary for YAML format in the context of keyword replacement
+// to preserve the keyword markers while also maintaining valid YAML syntax.
+export function wrapArrayReplaceMarkersInStrings(body: string, mappings: KeywordMappings): string {
```

Comment:
This function does the heavy lifting in this PR. It solves the problem outlined in Problem 1 of the description.